File: backend/app/schemas/automations.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field


class AutomationOut(BaseModel):
    id: int
    key: str
    name: str
    schedule_cron: Optional[str] = None
    is_enabled: bool
    params_json: dict[str, Any]
    last_run_at: Optional[datetime] = None
    next_run_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class AutomationRunOut(BaseModel):
    id: int
    automation_id: int
    started_at: datetime
    finished_at: Optional[datetime] = None
    status: str
    summary_json: dict[str, Any]
    error_text: Optional[str] = None

    class Config:
        from_attributes = True


class AutomationRunRequest(BaseModel):
    simulation: bool = True


class AutomationUpdate(BaseModel):
    name: Optional[str] = None
    schedule_cron: Optional[str] = None
    is_enabled: Optional[bool] = None
    params_json: Optional[dict[str, Any]] = None


class AutomationCreate(BaseModel):
    name: str
    schedule_cron: Optional[str] = None
    is_enabled: bool = True
    params_json: dict[str, Any] = Field(default_factory=dict)


class AutomationRunWithAutomation(BaseModel):
    id: int
    automation_id: int
    automation_name: str
    automation_key: str
    started_at: datetime
    finished_at: Optional[datetime] = None
    status: str
    summary_json: dict[str, Any]
    error_text: Optional[str] = None

File: frontend/src/api/automations.ts
import api from './client';

export interface Automation {
  id: number;
  key: string;
  name: string;
  schedule_cron?: string | null;
  is_enabled: boolean;
  params_json: Record<string, any>;
  last_run_at?: string | null;
  next_run_at?: string | null;
}

export interface AutomationRun {
  id: number;
  automation_id: number;
  automation_name: string;
  automation_key: string;
  started_at: string;
  finished_at?: string | null;
  status: string;
  summary_json: Record<string, any>;
  error_text?: string | null;
}

export const listAutomations = async () => {
  const { data } = await api.get<Automation[]>('/automations');
  return data;
};

export const runAutomation = async (id: number, simulation = true) => {
  const { data } = await api.post(`/automations/${id}/run`, { simulation });
  return data;
};

export const listAutomationRuns = async () => {
  const { data } = await api.get<AutomationRun[]>('/automations/runs');
  return data;
};
